components/scream/src/physics/rrtmgp/atmosphere_radiation.cpp

```diff
@@ -263,6 +263,10 @@ void RRTMGPRadiation::initialize_impl(const RunType /* run_type */) {
   // Determine whether or not we are using a fixed solar zenith angle (positive value)
   m_fixed_solar_zenith_angle = m_params.get<Real>("Fixed Solar Zenith Angle", -9999);
 
+  // Whether or not to do MCICA subcolumn sampling
+  m_do_subcol_sampling = m_params.get<bool>("do_subcol_sampling",false);
+  EKAT_REQUIRE_MSG(not m_do_subcol_sampling, "Error! RRTMGP does not yet support do_subcol_sampling = true");
```

Comment:
What is the logic that determines the error message? If `m_do_subcol_sampling` is `not` true, then write out the message? That doesn't seem right, so I must be missing something.

Reply:
ahhh, I think you are right @crterai , I'm surprised this would pass any tests without throwing an error right away.

Reply:
Ugg, now I'm so confused, but I think Ben is right.  So if `m_do_subcol_sampling` is FALSE than `not m_do_subcol_sampling` would be TRUE.  The REQUIRE does nothing if the input is TRUE.  So yeah, I take back my previous comment.

Reply:
```
#define IMPL_THROW(condition, msg, exception_type)                      \
  do {                                                                  \
    if ( ! (condition) ) {                                              \
      std::stringstream _ss_;                                           \
      _ss_ << __FILE__ << ":" << __LINE__ << ": FAIL:\n" << #condition; \
      _ss_ << "\n" << msg;                                              \
      throw exception_type(_ss_.str());                                 \
    }                                                                   \
  } while(0)

#define EKAT_REQUIRE_MSG(condition, msg)              IMPL_THROW(condition, msg, std::logic_error)
```

Reply:
If this is too confusing, we could change the flag to something like `disable_subcol_sampling`, so the assertion would look like
```
EKAT_REQUIRE_MSG(disable_subcol_sampling, "ERROR! Need to disable subcol sampling for now.");
```
Or, we could take subcol sampling out of it and use a flag that just says `use_gridmean_clouds` and then the assertion would be
```
EKAT_REQUIRE_MSG(use_gridmean_clouds, "ERROR: need to use grid-mean clouds for cloud optics until MCICA sampling is implemented");
```

Reply:
Ahh, I understand now. It throws an error if the first condition is not satisfied. Then it looks good to me.

Reply:
I think this is good, I think I just confused myself :-)